### PR TITLE
refactor: get module

### PR DESCRIPTION
**Pull request by MarcGonzalezMoratona**
In this pull request we are refactoring all the calls to GetModule in order to improve performance and code readability

### CODE REVIEW

File: Source/Modules/ModuleRender.cpp [hunk 1]
```diff
@@ -192,29 +193,35 @@ update_status ModuleRender::Update()
...
+		camera->GetCamera()->GetProjectionMatrix(), w, h);
 
-	GameObject* goSelected = App->GetModule<ModuleScene>()->GetSelectedGameObject();
+	GameObject* goSelected = scene->GetSelectedGameObject();
 
 	bool isRoot = goSelected->GetParent() == nullptr;
 
-	FillRenderList(App->GetModule<ModuleScene>()->GetLoadedScene()->GetRootQuadtree());
-	std::vector<GameObject*> nonStaticsGOs = App->GetModule<ModuleScene>()->GetLoadedScene()->GetNonStaticObjects();
+	FillRenderList(scene->GetLoadedScene()->GetRootQuadtree());
```
Comment (gerard-queralt) [inline Source/Modules/ModuleRender.cpp:223]: Maybe we could take this chance to remove some of this other repetitive calls, not only to the modules. There are a few examples, I'll point out some of them.
```suggestion
    Scene* loadedScene = scene->GetLoadedScene()
	FillRenderList(loadedScene->GetRootQuadtree());
```

File: Source/Modules/ModuleRender.cpp [hunk 2]
```diff
@@ -223,12 +230,12 @@ update_status ModuleRender::Update()
 #ifdef ENGINE
 	if (App->IsOnPlayMode())
 	{
-		AddToRenderList(App->GetModule<ModulePlayer>()->GetPlayer());
+		AddToRenderList(player->GetPlayer());
```
Comment (gerard-queralt) [inline Source/Modules/ModuleRender.cpp:233]:
```suggestion
		AddToRenderList(playerObject);
```

File: Source/Modules/ModuleRender.cpp [hunk 3]
```diff
@@ -341,9 +348,10 @@ bool ModuleRender::IsSupportedPath(const std::string& modelPath)
 
 void ModuleRender::FillRenderList(const Quadtree* quadtree)
 {
-	float3 cameraPos = App->GetModule<ModuleCamera>()->GetCamera()->GetPosition();
+	ModuleCamera* camera = App->GetModule<ModuleCamera>();
+	float3 cameraPos = camera->GetCamera()->GetPosition();
```
Comment (gerard-queralt) [inline Source/Modules/ModuleRender.cpp:352]:
```suggestion
    Camera* camera = moduleCamera->GetCamera();
	float3 cameraPos = camera->GetPosition();
```

File: Source/Modules/ModuleUI.cpp [hunk 4]
```diff
@@ -86,8 +89,8 @@ update_status ModuleUI::Update()
 	}
 
 	glEnable(GL_DEPTH_TEST);
-	App->GetModule<ModuleCamera>()->GetCamera()->GetFrustum()->
-		SetHorizontalFovAndAspectRatio(math::DegToRad(90), App->GetModule<ModuleCamera>()->GetCamera()->GetAspectRatio());
+	camera->GetCamera()->GetFrustum()->
```
Comment (gerard-queralt) [inline Source/Modules/ModuleUI.cpp:92]:
```suggestion
	camera->GetFrustum()->
```

File: Source/Physics/Physics.cpp [hunk 5]
```diff
@@ -78,12 +78,13 @@ bool Physics::ScreenPointToRay(const float2& mousePosition, LineSegment& ray)
...
 	std::map<float, const GameObject*> hitGameObjects;
+	ModuleScene* scene = App->GetModule<ModuleScene>();
 
 #ifdef ENGINE
-	AddIntersectionGameObject(hitGameObjects, ray, App->GetModule<ModuleScene>()->GetSelectedGameObject());
+	AddIntersectionGameObject(hitGameObjects, ray, scene->GetSelectedGameObject());
 #endif
-	AddIntersectionQuadtree(hitGameObjects, ray, App->GetModule<ModuleScene>()->GetLoadedScene()->GetRootQuadtree());
-	AddIntersectionDynamicObjects(hitGameObjects, ray, App->GetModule<ModuleScene>()->GetLoadedScene()->GetNonStaticObjects());
+	AddIntersectionQuadtree(hitGameObjects, ray, scene->GetLoadedScene()->GetRootQuadtree());
```
Comment (gerard-queralt) [inline Source/Physics/Physics.cpp:86]:
```suggestion
	AddIntersectionQuadtree(hitGameObjects, ray, loadedScene->GetRootQuadtree());
```

**Review by gerard-queralt — COMMENTED** [review]
Thank you Marc, left a suggestion to further reduce calls, but we can leave it for a separate task.

**Review by markmeliaa — APPROVED** [review]

**Review by gerard-queralt — APPROVED** [review]

**Review by markmeliaa — DISMISSED** [review]

**Review by Piterga12 — APPROVED** [review]
Good job, works great

**Review by DADSILVER — APPROVED** [review]

**Review by DADSILVER — APPROVED** [review]

**Review by gerard-queralt — APPROVED** [review]

**Comment by markmeliaa** [comment]
@MarcGonzalezMoratona if you fix the conflicts you could merge this.

**Review by diegusmejia117 — APPROVED** [review]

**Review by AuxPort66 — APPROVED** [review]

**Pull request merged**